lightning/src/routing/scoring.rs

```diff
@@ -722,6 +722,35 @@ impl<L: DerefMut<Target = u64>, T: Time, U: DerefMut<Target = T>> DirectedChanne
 	}
 }
 
+const FRACTIONAL_BITS: u32 = 3;
+const FRACTIONAL_BITMASK: u64 = (1 << FRACTIONAL_BITS) - 1;
+const LOG2_FRACTIONAL_PART: [f64; 1 << FRACTIONAL_BITS] =
+	[0.0, 0.17, 0.32, 0.46, 0.58, 0.70, 0.81, 0.91];
+const LOG2_10: f64 = 3.32;
+
+fn log10_approx(numerator: u64, denominator: u64) -> f64 {
+	(log2_approx(numerator) - log2_approx(denominator)) / LOG2_10
+}
+
+#[inline]
+fn log2_approx(x: u64) -> f64 {
+	let leading_zeros = x.leading_zeros();
+	let integer_part = (63 - leading_zeros) as f64;
+	let fractional_part = LOG2_FRACTIONAL_PART[
+		(((x << leading_zeros) >> (63 - FRACTIONAL_BITS)) & FRACTIONAL_BITMASK) as usize
```

Comment:
This commit kinda adds a good amount of complexity imo... Are we sure the perf gainz are worthwhile?

Reply:
My initial reaction is that those gains seem somewhat not worth it or at least better suited for follow-up, but would have to get @TheBlueMatt's thoughts

Reply:
Hmm, this seems....not useful at all? Its maybe useful but clearly dwarfed by noise, so unclear if useful. I'd say avoid it until a followup PR?